etc/nginx/nginx.conf

```diff
@@ -32,60 +32,57 @@ http {
 
     #gzip  on;
     client_max_body_size ${NGINX_CLIENT_MAX_BODY_SIZE};
-
-    server {
-        listen 80;
-        listen [::]:80;
-
-        server_name _ default;
-
-        root /var/www/localhost/htdocs;
-
-        index index.php;
-        
-        location ~ ^/(status|ping)$ {
-            log_subrequest off;
-            access_log off;
-            allow 127.0.0.1;
-            deny all;
-            include fastcgi_params;
-            fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
-            fastcgi_pass unix:/run/php-fpm.sock;
-        }
-
-        location / {
-            try_files $uri $uri/ /index.php?$args;
-        }
-
-        location ~* \.(?:ico|css|js|gif|jpe?g|png|ttf|woff)$ {
-            access_log off;
-            expires 30d;
-            add_header Pragma public;
-            add_header Cache-Control "public, mustrevalidate, proxy-revalidate";
-        }
-        
-        location ~ \.php$ {
-            try_files $uri =404;
-            
-            fastcgi_index index.php;
-            fastcgi_split_path_info ^(.+\.php)(.*)$;
-            fastcgi_keep_conn on;
-            include fastcgi_params;
-            fastcgi_param PATH_INFO       $fastcgi_path_info;
-            fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
-            fastcgi_pass unix:/run/php-fpm.sock;
-        }
-
-        location ~ /\.ht {
-            deny all;
-        }
-
-        location /protected {
-            deny all;
-        }
-
-        location /uploads/file {
-            deny all;
-        }
-    }
+	
+	server {
+		listen 80;
+		listen [::]:80;
+
+		server_name _ default;
+
+		root /var/www/localhost/htdocs;
+
+		index index.php;
+		
+		location / {
+			try_files $uri $uri/ __HUMHUB_SUB_DIR__/index.php?$args;
+			if ($query_string ~ "^r=content(/|%2)perma&id=([0-9]*)$") {
```

Comment:
could you elaborate on this? Is this a fix for some problem regarding perma-links or is it a workaround needed for subdir hosting? Maybe this should be added to the master if it makes sense to the "normal" installation as well.

Reply:
This is just a fix for the sub-dir installation. I'm no nginx wizard so I wouldn't be surprised if it broke things anyway.